scionlab/tests/test_user_as_models.py

```diff
@@ -574,108 +689,153 @@ def test_vpn_client_next_ip(self):
         former_vpn_ip = ip_address(vpn_client.ip)
         vpn_client.ip = str(former_vpn_ip + 1)
         vpn_client.save()
-        user_as = create_and_check_useras(self,
-                                          attachment_point=attachment_point,
-                                          owner=get_testuser(),
-                                          use_vpn=True,
-                                          public_port=50000)
+        aps_confs_new = _get_random_aps_params(seed, [attachment_point], VPNChoice.ALL)
+        user_as = create_and_check_random_useras(self,
+                                                 seed,
+                                                 aps_confs=aps_confs_new,
+                                                 vpn_choice=VPNChoice.ALL,
+                                                 owner=get_testuser())
         vpn_client_new = user_as.hosts.get().vpn_clients.get()
         self.assertEqual(ip_address(vpn_client_new.ip), former_vpn_ip)
 
-    @parameterized.expand(zip(range(testtopo_num_attachment_points)))
-    def test_change_ap(self, ap_index):
+    @parameterized.expand(zip(combinations(get_all_random_as_defs(), 2)))
+    def test_change_ap(self, AS_defs_pair):
         seed = 4
-        attachment_points = list(AttachmentPoint.objects.all())
-        user_as = create_random_useras(self,
-                                       seed=seed,
-                                       attachment_point=attachment_points[ap_index],
-                                       use_vpn=False)
-
-        other_ap_index = (ap_index + 1) % testtopo_num_attachment_points
-        self._change_ap(user_as, attachment_points[other_ap_index])
+        vpn_choice = VPNChoice.NONE
+        aps_pair = [as_defs2aps(AS_defs_pair[0]), as_defs2aps(AS_defs_pair[1])]
+        aps_confs = _get_random_aps_params(seed, aps_pair[0], vpn_choice)
+        user_as = create_and_check_random_useras(self,
+                                                 seed,
+                                                 aps_confs,
+                                                 vpn_choice,
+                                                 owner=get_testuser())
+
+        # Disable the previously set attachment points
+        for ap_conf in aps_confs:
+            ap_conf.active = False
```

Comment:
This is an interesting thing to check, but it's an entirely different test from what was here before. Can you please keep the test that just changes the attachment point selection?